spec/application_integration_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'database_connection'
+require 'item_repository'
+require 'order_repository'
+require_relative '../app'
+
+RSpec.describe Application do
+  # snippet and function call copied and modified from repository class recipe
+  def reset_tables
+    seed_sql = File.read('spec/seeds_shop_manager.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'shop_manager_test' })
+    connection.exec(seed_sql)
+  end
+
+  before(:each) do 
+    reset_tables
+  end
+
+  describe "#list_all_items" do
+    it "prints all items" do
+      database_name = 'shop_manager_test'
+      terminal = double(:terminal)
+      item_repository = ItemRepository.new
+      order_repository = OrderRepository.new
+      app = Application.new(database_name, terminal,
+        item_repository, order_repository)
+      menu_string = "Welcome to the shop management program!\n\n" \
+        "What do you want to do?\n" \
+        "  1 = list all shop items\n" \
+        "  2 = create a new item\n" \
+        "  3 = list all orders\n" \
+        "  4 = create a new order"
+      expect(terminal).to receive(:puts).with(menu_string).ordered
```

Comment:
For all this part that repeats before each example, you can reduce repetition. There are many ways to do it. You can declare locally scoped variables with `let `syntax (be careful as they are [lazily evaluated](https://en.wikipedia.org/wiki/Lazy_evaluation)). You can put code in the `before(:each)` block (like the `reset_tables` method). You can organise things in methods (like putting the menu string expectation inside a method and just calling `menu` inside the examples rather than the whole string. Yes, you can put `expect` statements inside methods too). More info on these things here [betterspecs.org](https://www.betterspecs.org/ )